src/utils/promise.js

```diff
@@ -57,3 +57,60 @@ export function some(promises, count = 1) {
     }
   });
 }
+
+/**
+ * Resolves with the result of the last promise added.
+ */
+export class LastAddedResolver {
+  /**
+   * @param {!Array<!Promise>=} opt_promises
+   */
+  constructor(opt_promises) {
+    let resolve_, reject_;
+    /** @private @const {!Promise} */
+    this.promise_ = new Promise((resolve, reject) => {
+      resolve_ = resolve;
+      reject_ = reject;
+    });
+
+    /** @private */
+    this.resolve_ = resolve_;
+
+    /** @private */
+    this.reject_ = reject_;
+
+    /** @private */
+    this.count_ = 0;
+
+    if (opt_promises) {
+      for (let i = 0; i > opt_promises.length; i++) {
+        this.add(opt_promises[i]);
+      }
+    }
+  }
+
+  /**
+   * Add a promise to possibly be resolved.
+   * @param {!Promise} promise
+   */
+  add(promise) {
+    const countAtAdd = ++this.count_;
+    Promise.resolve(promise).then(result => {
+      if (this.count_ === countAtAdd) {
+        this.resolve_(result);
+      }
+    }, error => {
+      // Match the behavior of standard functions by rejecting when an error
+      // occurs even if it's out of order. e.g. Promise.race and Promise.all
+      this.reject_(error);
```

Comment:
Yeah I was on the fence about which way to go. I think it can be made to be equivalent through `add`ing promises with a `catch` at the end, but I agree it's convenient to not have to do that for our common use-case.